packages/common/src/registry.ts

```diff
@@ -0,0 +1,115 @@
+import ProjectRegistryABI from "./abis/ProjectRegistry.json";
+import { ethers } from "ethers";
+import { Provider } from "@wagmi/core";
+import { Web3Provider } from "@ethersproject/providers";
+import {
+  Multicall,
+  ContractCallContext,
+} from "ethereum-multicall";
+
+export const fetchProjectOwners = async (
+  provider: Provider | Web3Provider,
+  chainID: number,
+  projectID: number,
+): Promise<string[]> => {
+  const addresses = addressesByChainID(chainID);
+
+  const projectRegistry = new ethers.Contract(
+    addresses.projectRegistry!,
+    ProjectRegistryABI,
+    provider,
+  );
+
+  try {
+    const projectOwners = await projectRegistry.getProjectOwners(projectID);
+    return projectOwners as string[];
+  } catch (err) {
+    // console.log(err);
+    console.log(
+      `Error fetching owners from registry on chain ${chainID} for project ${projectID}`,
+    );
+    return [];
+  }
+};
+
+export const fetchMultipleProjectOwners = async (
+  provider: Provider | Web3Provider,
+  chainID: number,
+  projectIDs: number[],
+): Promise<string[][]> => {
+
+  const addresses = addressesByChainID(chainID);
+  const registryAddress = addresses.projectRegistry!;
+
+  const multicall = new Multicall({
+    ethersProvider: provider,
+    tryAggregate: true,
+  });
+
+  const contractCallContext: ContractCallContext[] = [
+    {
+      reference: "projectRegistry",
+      contractAddress: registryAddress,
+      abi: ProjectRegistryABI,
+      calls: [],
+    },
+  ]
+
+  for (const projectID of projectIDs) {
+    contractCallContext[0].calls.push({
+      reference: projectID.toString(),
+      methodName: "getProjectOwners",
+      methodParameters: [projectID],
+    });
+  }
+
+  const results: any = await multicall.call(
+    contractCallContext,
+  );
+
+  return results.results?.projectRegistry ? results.results.projectRegistry.callsReturnContext.map(
+    (result: any) => (result.returnValues ? result.returnValues : []),
+  ) : [[]];
+};
+
+export const addressesByChainID = (chainID: number) => {
+  const chainName: string = chains[chainID];
+  return addresses[chainName];
+};
+
+export const chains: { [key: number]: string } = {
```

Comment:
is there an enum available for this?

Reply:
Not yet, we could put this on the list for our refactor/tech-debt sprint and put some enums in the common package.